File: spelling_bee.pl
#!/usr/bin/env perl
use v5.36;
use autodie;

my $dict = '/usr/share/dict/american-english-huge';
my @c = split //, $ARGV[0];
my $middle = $c[0];
my $re = make_regex(@c);

open my $fh, '<', $dict;
while (<$fh>) {
    chomp;
    next unless length >= 4;
    next unless /$middle/;
    next if /$re/;
    say is_pangram($_) ? "$_ (pangram)" : $_;
}

sub make_regex(@c) {
    my $letters = join "", @c;
    return qr([^$letters]);
}

sub is_pangram($s) {
    return 0 if length($s) < 7;
    my %h = map {$_ => 1} split //, $s;
    return scalar keys %h == 7;
}

File: spelling_bee.py
from sys import argv
import re

def make_regex(c):
    return f'[^{c}]'

def is_pangram(s):
    if len(s) < 7:
        return False;
    return len({ c for c in s }) == 7

DICT = '/usr/share/dict/american-english-large';
c = argv[1]
middle = c[0]
regex = make_regex(c)

with open(DICT) as f:
    for line in f:
        line = line.rstrip()
        if len(line) < 4 or middle not in line or re.search(regex, line):
            continue
        print(f'{line} (pangram)' if is_pangram(line) else line)
